File: app/src/main/java/com/devidea/timeleft/TopRecyclerView.java
package com.devidea.timeleft;


import android.animation.ObjectAnimator;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

import static com.devidea.timeleft.MainActivity.ITEM_GENERATE;

class TopRecyclerView extends androidx.recyclerview.widget.RecyclerView.Adapter<TopRecyclerView.ViewHolder> {

    //array list
    private final ArrayList<AdapterItem> arrayList;

    //CustomAdapter 생성자
    public TopRecyclerView(ArrayList<AdapterItem> arrayList) {
        this.arrayList = arrayList;
    }

    // Create new views (invoked by the layout manager)
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
        // Create a new view, which defines the UI of the list item
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_recyclerview_top, viewGroup, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, int position) {

        //String summery_buf = arrayList.get(position).getSummery();
        viewHolder.summery.setText(arrayList.get(position).getSummery());

        String percent_buf = arrayList.get(position).getPercentString();
        viewHolder.percent.setText(percent_buf + "%");

        viewHolder.startValue.setText(arrayList.get(position).getStartDay());
        viewHolder.endValue.setText(arrayList.get(position).getEndDay());
        viewHolder.leftValue.setText(arrayList.get(position).getLeftDay());

        int percent = (int) Float.parseFloat(percent_buf);

        ObjectAnimator.ofInt(viewHolder.progressBar, "progress", percent)
                .setDuration(1500)
                .start();

    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position, @NonNull List<Object> payloads) {
        if (payloads.isEmpty()) {
            super.onBindViewHolder(holder, position, payloads);
        } else {
            AdapterItem adapterItem = ITEM_GENERATE.timeItem();
            holder.percent.setText(adapterItem.getPercentString() + "%");
            holder.progressBar.setProgress((int) Float.parseFloat(adapterItem.getPercentString()));
            holder.leftValue.setText(adapterItem.getLeftDay());
        }
    }

    @Override
    public int getItemCount() {
        return arrayList.size();
    }

    public static class ViewHolder extends androidx.recyclerview.widget.RecyclerView.ViewHolder {
        private final TextView summery;
        private final TextView percent;
        private final ProgressBar progressBar;
        private final TextView startValue;
        private final TextView endValue;
        private final TextView leftValue;

        //ViewHolder
        public ViewHolder(View view) {
            super(view);
            summery = (TextView) view.findViewById(R.id.summery);
            percent = (TextView) view.findViewById(R.id.percent_text);
            progressBar = (ProgressBar) view.findViewById(R.id.progress);

            startValue = view.findViewById(R.id.start_day);
            endValue = view.findViewById(R.id.end_day);
            leftValue = view.findViewById(R.id.left_day);

        }
    }
}

File: app/src/main/java/com/devidea/timeleft/TopRecyclerView.kt
package com.devidea.timeleft

import androidx.recyclerview.widget.RecyclerView
import android.widget.TextView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.animation.ObjectAnimator
import android.widget.ProgressBar
import android.view.View
import java.util.ArrayList

/**
이 TopRecyclerView 는 ViewHolder 의 Binding 을 통해 작성되었습니다.
 */

internal class TopRecyclerView
constructor(private val arrayList: ArrayList<AdapterItem>) :
    RecyclerView.Adapter<TopRecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val view: View = LayoutInflater.from(viewGroup.context)
            .inflate(R.layout.item_recyclerview_top, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {

        viewHolder.title.text = arrayList[position].title
        viewHolder.percent.text = arrayList[position].percent.toString() + "%"
        viewHolder.leftValue.text = arrayList[position].leftString
        ObjectAnimator.ofInt(viewHolder.progressBar, "progress", arrayList[position].percent.toInt())
            .setDuration(1500)
            .start()
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int, payloads: List<Any>) {

        if (payloads.isEmpty()) {
            super.onBindViewHolder(holder, position, payloads)
        } else {
            val adapterItem: AdapterItem = payloads[0] as AdapterItem

            holder.percent.text = adapterItem.percent.toString() + "%"
            holder.progressBar.progress = adapterItem.percent.toInt()
            holder.leftValue.text = adapterItem.leftString
        }
    }

    override fun getItemCount(): Int {
        return arrayList.size
    }

    class ViewHolder constructor(view: View) : RecyclerView.ViewHolder(view) {
        val title: TextView
        val percent: TextView
        val progressBar: ProgressBar
        val leftValue: TextView

        //ViewHolder
        init {
            title = view.findViewById<View>(R.id.title) as TextView
            percent = view.findViewById<View>(R.id.percent_text) as TextView
            progressBar = view.findViewById<View>(R.id.progress) as ProgressBar
            leftValue = view.findViewById(R.id.left_day)
        }
    }
}